modules/optflow/src/pcaflow.cpp

```diff
@@ -0,0 +1,508 @@
+    u += A * v;
+    beta = cv::norm( u, NORM_L2 );
+
+    if ( beta > 0 )
+    {
+      u *= 1 / beta;
+      v *= -beta;
+      v += AT * u;
+      alfa = cv::norm( v, NORM_L2 );
+      if ( alfa > 0 )
+        v = ( 1 / alfa ) * v;
+    }
+
+    double rhobar1 = sqrt( rhobar * rhobar + damp * damp );
+    double cs1 = rhobar / rhobar1;
+    phibar = cs1 * phibar;
+
+    double cs, sn, rho;
+    symOrtho( rhobar1, beta, cs, sn, rho );
+
+    double theta = sn * alfa;
+    rhobar = -cs * alfa;
+    double phi = cs * phibar;
+    phibar = sn * phibar;
+
+    double t1 = phi / rho;
+    double t2 = -theta / rho;
+
+    x += t1 * w;
+    w *= t2;
+    w += v;
+  }
+}
+
+void OpticalFlowPCAFlow::findSparseFeatures( UMat &from, UMat &to, std::vector<Point2f> &features,
+                                             std::vector<Point2f> &predictedFeatures ) const
+{
+  Size size = from.size();
+  const unsigned maxFeatures = size.area() * sparseRate;
+  goodFeaturesToTrack( from, features, maxFeatures * retainedCornersFraction, 0.005, 3 );
+
+  // Add points along the grid if not enough features
+  if ( maxFeatures > features.size() )
+  {
+    const unsigned missingPoints = maxFeatures - features.size();
+    const unsigned blockSize = sqrt( (float)size.area() / missingPoints );
+    for ( int x = blockSize / 2; x < size.width; x += blockSize )
+      for ( int y = blockSize / 2; y < size.height; y += blockSize )
+        features.push_back( Point2f( x, y ) );
+  }
+  std::vector<uchar> predictedStatus;
+  std::vector<float> predictedError;
+  calcOpticalFlowPyrLK( from, to, features, predictedFeatures, predictedStatus, predictedError );
+
+  size_t j = 0;
+  for ( size_t i = 0; i < features.size(); ++i )
+  {
+    if ( predictedStatus[i] )
+    {
+      features[j] = features[i];
+      predictedFeatures[j] = predictedFeatures[i];
+      ++j;
+    }
+  }
+  features.resize( j );
+  predictedFeatures.resize( j );
+}
+
+void OpticalFlowPCAFlow::removeOcclusions( UMat &from, UMat &to, std::vector<Point2f> &features,
+                                           std::vector<Point2f> &predictedFeatures ) const
+{
+  std::vector<uchar> predictedStatus;
+  std::vector<float> predictedError;
+  std::vector<Point2f> backwardFeatures;
+  calcOpticalFlowPyrLK( to, from, predictedFeatures, backwardFeatures, predictedStatus, predictedError );
+
+  size_t j = 0;
+  const float threshold = occlusionsThreshold * sqrt( from.size().area() );
+  for ( size_t i = 0; i < predictedFeatures.size(); ++i )
+  {
+    if ( predictedStatus[i] )
+    {
+      Point2f flowDiff = features[i] - backwardFeatures[i];
+      if ( flowDiff.dot( flowDiff ) <= threshold )
+      {
+        features[j] = features[i];
+        predictedFeatures[j] = predictedFeatures[i];
+        ++j;
+      }
+    }
+  }
+  features.resize( j );
+  predictedFeatures.resize( j );
+}
+
+static inline void _cpu_fillDCTSampledPoints( float *row, const Point2f &p, const Size &basisSize, const Size &size )
```

Comment:
Advise using anonymous namespace for these functions to hide linkage, rather than the static keyword.  More idiomatic c++.